### PR TITLE
Apache Flume Initial Package 1.9.0

### PR DESCRIPTION
**Pull request by polynomial**
###### Motivation for this change

We use Apache Flume in production and are migrating everything to NixOS. I will write a module next, but this gets me a working Apache Flume jar, which is a great first step!

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/servers/apache-flume/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+stdenv.mkDerivation rec {
+  version = "${flumeVersion}";
+  name = "apache-flume-${version}";
+
+  src = fetchurl {
+    url = "mirror://apache/flume/${version}/apache-flume-${version}-bin.tar.gz";
+    inherit sha256;
+  };
+
+  buildInputs = [ hadoop jre makeWrapper bash gnused coreutils ];
```
Comment (volth) [inline pkgs/servers/apache-flume/default.nix]: ```bash``` ```gnused``` ```coreutils``` are already included in every bulid by default

Comment (polynomial) [reply pkgs/servers/apache-flume/default.nix]: Fixed, thanks! Old habit's die hard (:

**Comment by polynomial** [comment]
What are the steps to get this merged? I have addressed the comment and am using this locally, would love to see it be part of the tree. I am currently working on a module and will submit a PR for that shortly.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -8580,6 +8580,9 @@ in
   apacheKafka_2_1 = callPackage ../servers/apache-kafka { majorVersion = "2.1"; };
   apacheKafka_2_2 = callPackage ../servers/apache-kafka { majorVersion = "2.2"; };
 
+  apacheFlume = apacheFlume_1_9_0;
+  apacheFlume_1_9_0 = callPackage ../servers/apache-flume { majorVersion = "1.9.0"; };
```
Comment (JohnAZoidberg) [inline pkgs/top-level/all-packages.nix]: Do you think there will be reasons to keep multiple versions of flume?
It does not seem like there are multiple maintained at the same time.
If that's not the case, it'd be better not to have to maintain multiple versions for us.

File: pkgs/servers/apache-flume/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      --set FLUME_HOME "$out" \
+      --set JAVA_HOME "${jre}"
+    chmod +x $out/bin\/*
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = http://flume.apache.org;
+    description = "Flume is a distributed, reliable, and available service for efficiently collecting, aggregating, and moving large amounts of log data.";
+    license = licenses.asl20;
+    maintainers = [ ];
```
Comment (JohnAZoidberg) [inline pkgs/servers/apache-flume/default.nix]: Would you be willing to look after this package, i.e. be maintainer?

File: pkgs/servers/apache-flume/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    mkdir -p $out/bin
+    cp bin/flume-ng $out/bin
+    wrapProgram $out/bin/flume-ng \
+      --set FLUME_HOME "$out" \
+      --set JAVA_HOME "${jre}"
+    chmod +x $out/bin\/*
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = http://flume.apache.org;
```
Comment (JohnAZoidberg) [inline pkgs/servers/apache-flume/default.nix]:
```suggestion
    homepage = "https://flume.apache.org";
```
We'll deprecate unquoted URL syntax, once RFC 45 is approved. Also their site supports https.

File: pkgs/servers/apache-flume/default.nix [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  installPhase = ''
+    mkdir -p $out
+    cp -R tools conf lib $out
+
+    mkdir -p $out/bin
+    cp bin/flume-ng $out/bin
+    wrapProgram $out/bin/flume-ng \
+      --set FLUME_HOME "$out" \
+      --set JAVA_HOME "${jre}"
+    chmod +x $out/bin\/*
```
Comment (JohnAZoidberg) [inline pkgs/servers/apache-flume/default.nix]: I'm pretty sure `wrapProgram` already marks its output as executable. Please try to remove it.

File: pkgs/servers/apache-flume/default.nix [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      sha256 = "069nnakibv4pdra06fnq08nb7qypzx8j586r4fmw8kflpxdfswq3";
+    };
+  };
+in
+
+with versionMap.${majorVersion};
+
+stdenv.mkDerivation rec {
+  version = "${flumeVersion}";
+  name = "apache-flume-${version}";
```
Comment (JohnAZoidberg) [inline pkgs/servers/apache-flume/default.nix]:
```suggestion
  pname = "apache-flume";
```

`pname` appends the version name on its own.

File: pkgs/servers/apache-flume/default.nix [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+      flumeVersion = "1.9.0";
+      sha256 = "069nnakibv4pdra06fnq08nb7qypzx8j586r4fmw8kflpxdfswq3";
+    };
+  };
+in
+
+with versionMap.${majorVersion};
+
+stdenv.mkDerivation rec {
+  version = "${flumeVersion}";
```
Comment (JohnAZoidberg) [inline pkgs/servers/apache-flume/default.nix]:
```suggestion
  version = flumeVersion;
```
No string quoting necessary here :)

File: pkgs/servers/apache-flume/default.nix [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    cp bin/flume-ng $out/bin
+    wrapProgram $out/bin/flume-ng \
+      --set FLUME_HOME "$out" \
+      --set JAVA_HOME "${jre}"
+    chmod +x $out/bin\/*
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = http://flume.apache.org;
+    description = "Flume is a distributed, reliable, and available service for efficiently collecting, aggregating, and moving large amounts of log data.";
```
Comment (JohnAZoidberg) [inline pkgs/servers/apache-flume/default.nix]: Please follow https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes

**Comment by SuperSandro2000** [comment]
Closing due to being stale.

**Pull request closed without merging**